app/models/alchemy/element/dom_id.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Alchemy
+  # Returns a dom id used for elements html id tag.
+  #
+  # Uses the elements name and its position on the page.
+  # If the element is nested in a parent element it prefixes
+  # the id with the parent elements dom_id.
+  #
+  # Register your own dom id class with
+  #
+  #  Alchemy::Element.dom_id_class = MyDomIdClass
+  #
+  class Element < BaseRecord
+    class DomId
+      def initialize(element)
+        @element = element
+        @parent_element = element.parent_element
+      end
+
+      def call
+        [parent_element&.dom_id, element.name, element.position].compact.join("-")
```

Comment:
Using the parent element's DOM ID here can lead to recursion and a very long DOM ID for a deeply nested element; maybe it's simpler to just use the parent element's ID.

Reply:
See #2354 for why the ID does not work.

And this comment is exactly why I introduced this class. If you think the dom id gets too long, implement your own logic. I don't think we can and should solve this in Alchemy